syntax = "proto3";

package audio_streaming;

// Audio streaming service definition
service AudioStreamer {
  // Stream audio data from client to server
  rpc StreamAudio (stream AudioRequest) returns (AudioResponse) {}
}

// Audio request message containing the audio chunk
message AudioRequest {
  bytes audio_chunk = 1;  // Raw audio data chunk
  int32 sample_rate = 2;  // Sample rate of the audio
  int32 channels = 3;     // Number of audio channels
  string format = 4;      // Audio format (e.g., "int16", "float32")
}

// Response from the server
message AudioResponse {
  bool success = 1;       // Whether streaming was successful
  string message = 2;     // Response message
}